GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -65,15 +113,18 @@ protected override void PerformMaintenance()
                         }
                     }
 
-                    int beforeCount = this.CountLooseObjects();
+                    int beforeLooseObjectsCount = this.CountLooseObjects();
                     GitProcess.Result result = this.RunGitCommand((process) => process.PrunePacked(this.Context.Enlistment.GitObjectsRoot));
-                    int afterCount = this.CountLooseObjects();
+                    int afterLooseObjectsCount = this.CountLooseObjects();
+
+                    this.CreateLooseObjectsPackFile();
 
                     EventMetadata metadata = new EventMetadata();
                     metadata.Add("GitObjectsRoot", this.Context.Enlistment.GitObjectsRoot);
-                    metadata.Add("StartingCount", beforeCount);
-                    metadata.Add("EndingCount", afterCount);
-                    metadata.Add("RemovedCount", beforeCount - afterCount);
+                    metadata.Add("StartingCount", beforeLooseObjectsCount);
+                    metadata.Add("EndingCount", afterLooseObjectsCount);
+                    metadata.Add("RemovedCount", beforeLooseObjectsCount - afterLooseObjectsCount);
+                    metadata.Add("LooseObjectsPutIntoPackFile", this.LooseObjectsPutIntoPackFile);

```

Comment:
Now that is an exellent find!  This would cause a problem for sure.  I'll add the reset.  
It's a member variable so can unit test it.  If you see an easier way to do that let me know!

Reply:
You could also return it from `CreateLooseObjectsPackFile()` and avoid the member.

We create a new instance of the step each time we queue it, so it's not a bug at the moment. (Could creep in if we changed that.)

Reply:
> It's a member variable so can unit test it. If you see an easier way to do that let me know!

I was thinking an approach like this would allow the member variable to be removed:

https://github.com/wilbaker/VFSForGit/commit/3405d9bd55786ae861a0004fb4b05ece017cf71e

(You could also use `out` parameters rather than returning the number of loose objects added to the pack).

The advantage I see to that approach is there's no risk making a bad decision based on the value of `this.LooseObjectsPutIntoPackFile`

Reply:
Note: I confirmed the unit tests passed with my changes, but I did not validate the correct loose object count is being written to the log.

Reply:
> We create a new instance of the step each time we queue it, so it's not a bug at the moment. (Could creep in if we changed that.)

Ahh good call, I missed that detail

Reply:
I'll happily cherry-pick that commit over @wilbaker thanks!